instrumentation/opentelemetry-instrumentation-cherrypy/tests/test_cherrypy.py

```diff
@@ -0,0 +1,513 @@
+            "http.request.header.custom_test_header_2": (
+                "test-header-value-2",
+            ),
+        }
+        resp = self.call(
+            url="/check_header",
+            headers=[
+                ("custom-test-header-1","test-header-value-1"),
+                ("custom-test-header-2","test-header-value-2"),
+            ],
+        )
+        self.assertEqual('200 OK', resp[0])
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+        server_span = [
+            span for span in span_list if span.kind == trace.SpanKind.SERVER
+        ][0]
+
+        self.assertSpanHasAttributes(server_span, expected)
+    
+    def test_http_custom_request_headers_not_in_span_attributes(self):
+        not_expected = {
+            "http.request.header.custom_test_header_3": (
+                "test-header-value-3",
+            ),
+        }
+        resp = self.call(
+            url="/check_header",
+            headers=[
+                ("custom-test-header-1","test-header-value-1"),
+                ("custom-test-header-2","test-header-value-2"),
+            ],
+        )
+        self.assertEqual('200 OK', resp[0])
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+        server_span = [
+            span for span in span_list if span.kind == trace.SpanKind.SERVER
+        ][0]
+
+        for key, _ in not_expected.items():
+            self.assertNotIn(key, server_span.attributes)
+
+    def test_http_custom_response_headers_in_span_attributes(self):
+        expected = {
+            "http.response.header.custom_test_header_1": (
+                "test-header-value-1",
+            ),
+            "http.response.header.custom_test_header_2": (
+                "test-header-value-2",
+            ),
+        }
+        resp = self.call(url="/check_header")
+        self.assertEqual('200 OK', resp[0])
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+        server_span = [
+            span for span in span_list if span.kind == trace.SpanKind.SERVER
+        ][0]
+        self.assertSpanHasAttributes(server_span, expected)
+
+    def test_http_custom_response_headers_not_in_span_attributes(self):
+        not_expected = {
+            "http.response.header.custom_test_header_3": (
+                "test-header-value-3",
+            ),
+        }
+        resp = self.call(url="/check_header")
+        self.assertEqual('200 OK', resp[0])
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+        server_span = [
+            span for span in span_list if span.kind == trace.SpanKind.SERVER
+        ][0]
+
+        for key, _ in not_expected.items():
+            self.assertNotIn(key, server_span.attributes)
+
+class TestNonRecordingSpanWithCustomHeaders(TestBase, helper.CPWebCase):
```

Comment:
why dont you inherit this class from the TestCherryPyBase and override all functions? This will remove some redundant code